snapcraft/internal/review_tools/_runner.py

```diff
@@ -0,0 +1,50 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import pathlib
+import subprocess
+
+from . import errors
+
+
+def is_available() -> bool:
+    review_tools_bin = pathlib.Path("/snap/bin/review-tools.snap-review")
+    return review_tools_bin.exists()
+
+
+def run(*, snap_filename: str) -> None:
+    review_tools_bin = pathlib.Path("/snap/bin/review-tools.snap-review")
+    if not review_tools_bin.exists():
+        raise errors.ReviewToolNotAvailableError()
+
+    # TODO allow suppression of error and warn through project configuration.
+    # mypy misses out on pathlib.Path, so we convert it
+    command = [review_tools_bin.as_posix(), snap_filename, "--json"]
+    # Speed up the process by not doing the re-squash tests.
+    env = dict(SNAP_ENFORCE_RESQUASHFS="0")
+
+    try:
+        subprocess.check_output(command, stderr=subprocess.STDOUT, env=env)
+    except subprocess.CalledProcessError as call_error:
+        # Error codes:
+        # 2 -> warnings.
+        # 3 -> warnings and errors.
+        # Any of these trigger a store review.
+        if call_error.returncode not in (2, 3):
+            raise
+        # We only really care about v2_lint now.
+        raise errors.ReviewError(review_json=json.loads(call_error.output.decode()))
```

Comment:
I assume the presence of v2_lint is guaranteed with a return code of 2/3?

Reply:
yes, the payload is always there, @jdstand can confirm